sci-physics/reduze/files/reduze-2.0.9-gcc6.patch

```diff
@@ -0,0 +1,58 @@
+--- a/reduze/amplitude.cpp
++++ b/reduze/amplitude.cpp
+@@ -782,7 +782,7 @@
+ 
+ 	ex resultLeadingNc = result;
+ 	for (exset::const_iterator it = found.begin(); it != found.end(); ++it) {
+-		VERIFY(is_a<function>(*it) && ex_to<function>(*it).get_name() == "Color");
++		VERIFY(is_a<GiNaC::function>(*it) && ex_to<GiNaC::function>(*it).get_name() == "Color");
+ 		VERIFY(is_a<lst>(it->op(0)));
+ 		ex res(1);
+ 		for (size_t i = 0; i < it->op(0).nops(); ++i)
+@@ -823,7 +823,7 @@
+ 	timer.restart();
+ 	exmap m;
+ 	for (exset::const_iterator it = found.begin(); it != found.end(); ++it) {
+-		VERIFY(is_a<function>(*it) && ex_to<function>(*it).get_name() == "Lorentz");
++		VERIFY(is_a<GiNaC::function>(*it) && ex_to<GiNaC::function>(*it).get_name() == "Lorentz");
+ 		VERIFY(is_a<lst>(it->op(0)));
+ 
+ 		ex res(1);
+--- a/reduze/files.cpp
++++ b/reduze/files.cpp
+@@ -420,7 +420,7 @@
+ 			get_filename_sectormappings(m.integralfamily()->name());
+ 	ofstream file(file_name.c_str());
+ 	if (!file)
+-		ABORT("Can't write to " << file);
++		ABORT("Can't write to " << file_name);
+ 	os << YAML::BeginMap << YAML::Key << "sectormappings" << YAML::Value;
+ 	os << m << YAML::EndMap;
+ 	file << os.c_str();
+@@ -617,7 +617,7 @@
+ 
+ bool is_readable_file(const string& filename) {
+ 	ifstream is(filename.c_str());
+-	bool ok = is;
++	bool ok = !is.fail();
```

Comment:
Again, `is` would be casted to `void *` and then to `bool` in C++98.  `!is.fail()` has the same effect and is dialect neutral.